### PR TITLE
[opt](merge-on-write) Skip the alignment process of some rowsets in partial update

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

If a rowset is produced by compaction before the commit phase of the partial update load and is not included in `txn_info->rowset_ids`, we can skip the alignment process of that rowset because data remains the same before and after compaction. But we still need to calculate the the delete bitmap for that rowset.

branch-2.1-pick: https://github.com/apache/doris/pull/38682
branch-2.0-pick: https://github.com/apache/doris/pull/38686

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

File: be/src/olap/base_tablet.cpp [hunk 1]
```diff
@@ -1242,6 +1244,33 @@ Status BaseTablet::update_delete_bitmap(const BaseTabletSPtr& self, TabletTxnInf
...
 
+    // If a rowset is produced by compaction before the commit phase of the partial update load
+    // and is not included in txn_info->rowset_ids, we can skip the alignment process of that rowset
+    // because data remains the same before and after compaction. But we still need to calculate the
+    // the delete bitmap for that rowset.
+    std::vector<RowsetSharedPtr> rowsets_skip_alignment;
+    if (is_partial_update) {
+        std::vector<RowsetSharedPtr> remained_rowsets;
+        for (const auto& rowset : specified_rowsets) {
+            if (rowset->produced_by_compaction()) {
```
Comment (zhannngchen) [inline be/src/olap/base_tablet.cpp:1255]: should consider that it's end version is less or equal to the max version in txn_info->rowset_ids.

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Comment by bobhan1** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by hust-hhb — APPROVED** [review]
LGTM

**Pull request merged**